fdb-extensions/src/main/java/com/apple/foundationdb/async/MoreAsyncUtil.java

```diff
@@ -857,6 +857,33 @@ public static void closeIterator(@Nonnull Iterator<?> iterator) {
         }
     }
 
+    /**
+     * Compose a handler bi-function to the result of a future. Unlike the
+     * {@link AsyncUtil#composeHandle(CompletableFuture, BiFunction)}, which completes exceptionally only when the
+     * <code>handler</code> completes exceptionally, it completes exceptionally even if the supplied action itself
+     * (<code>future</code>) encounters an exception.
```

Comment:
Perhaps something like, "This is supposed to replicate the semantics of `CompletionStage#whenComplete` but to handle the case where the completion handler might itself contain async work."

Reply:
Added as the doc of the new `composeWhenComplete`.